### DISCUSSION TITLE
Include 'private_hash' of file in the collection API return values

### DISCUSSION THREAD

**Discussion by wastelandtime** [directus/directus#11369] [Archive]
One of the fields of a collection is an image. The field that is being returned for that ('featured_image') is a number identifying the image.

In order to get the image/thumbnail url I have to make a separate call to the 'files' endpoint. 
Is there a way of including the image 'private_hash' or even better a full image/thumbnail url within the collection call to the REST API?

My call to the collection:

`https://cms.example.com/project_name/items/collection_name?filter[featured_image][nnull]`

Thanks

**Comment by benhaynes** [comment]
Use the fields parameter:
https://docs.directus.io/reference/query/#fields

**Reply by wastelandtime** [reply]
Thanks. I don't seem to be able to combine both the filter and the fields parameter. If I remove the filter, and leave the fields parameter it works fine but I'd need the filter there as well. Is it possible to use both at the same time?

`https://cms.example.com/project_name/items/collection_name?filter[my_field][nempty]?fields=*.private_hash
`
`https://cms.example.com/project_name/items/collection_name?fields=*.private_hash?filter[my_field][nempty]`

**Reply by benhaynes** [reply]
You should do a bit more research into how query parameters work. You use a `?` first, and then `&` for other parameters.

**Reply by wastelandtime** [reply]
Thanks a lot!
